api/agent/caasapplication/client.go

```diff
@@ -42,6 +43,11 @@ func (c *Client) UnitIntroduction(podName string, podUUID string) (*UnitConfig,
 		return nil, err
 	}
 	if err := result.Error; err != nil {
+		if params.IsCodeAlreadyExists(err) {
+			return nil, errors.AlreadyExists
+		} else if params.IsCodeNotAssigned(err) {
+			return nil, errors.NotAssigned
+		}
```

Comment:
We could have used `apiservererrors.RestoreError()` here

Reply:
Will handle in 2.9